### PR TITLE
Fix alignment issue

### PR DESCRIPTION
**Pull request by jcarpent**
Related to #1201

@cmastalli I think this PR should solve your issue.

### CODE REVIEW

**Comment by cmastalli** [comment]
@jcarpent that's quick fast response. Thank you so much ;)

Do you want me to try?

**Comment by jcarpent** [comment]
Yes, you can try it.

**Comment by cmastalli** [comment]
Are you sure this solve the issue?

In my case, I am still having the same problem.

File: bindings/python/multibody/data.hpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include <eigenpy/memory.hpp>
 #include <eigenpy/exception.hpp>
 
+#include "pinocchio/bindings/python/multibody/model.hpp" // Requested for memory alignment issues
```
Comment (cmastalli) [inline bindings/python/multibody/data.hpp:16]: Probably you need to change the inclusion order. It seems to me that we need to include first `std-vector.hpp` and `std-aligned-vector.hpp`, right?

Comment (cmastalli) [reply bindings/python/multibody/data.hpp:16]: I tried it locally, and it doesn't solve neither the issue in my PC.

Comment (jcarpent) [reply bindings/python/multibody/data.hpp:16]: No, we don't care. They are not involved in your script.

**Comment by jcarpent** [comment]
> Are you sure this solve the issue?
> 
> In my case, I am still having the same problem.
 Do you have reinstall pinocchio? Are you sure the correct version has been installed?

**Comment by cmastalli** [comment]
> Do you have reinstalled pinocchio? Are you sure the correct version has been installed?

I uninstalled my previous Pinocchio library, and then I tried directly using the `libpinocchio_pywrap.so` generated inside the build folder.

**Comment by jcarpent** [comment]
Alignment should then be enforced for all the parsers then.

**Comment by jcarpent** [comment]
@cmastalli Could try with an empty model. I suspect the issue is coming from `buildSampleModelHumanoid` and related function that returns a model or take a model as input. Could you also try `pin.neutral(model)` in the vein.

**Comment by cmastalli** [comment]
> @cmastalli Could try with an empty model

it is still failing with the following code:
```python
import pinocchio
model = pinocchio.Model()
for i in range(100):
  data = pinocchio.Data(model)
```



> Could you also try `pin.neutral(model)` in the same vein?

It works for both cases: empty and sample humanoid models.

**Comment by jcarpent** [comment]
I've checked, and on my side, I don't have any issue. Everything works well.

**Comment by cmastalli** [comment]
> I've checked, and on my side, I don't have any issue. Everything works well.

Are you using Eigen 3.3.7? I will try it in another PC.

**Comment by jcarpent** [comment]
Yep.

**Comment by cmastalli** [comment]
@jcarpent I have tried in another PC with the same conditions 
 - Release mode and `-march=native` for Pinocchio and Eigenpy,
 - Eigen 3.3.7

And, unfortunately, I found again the same issue. Could be this related to the compiler? (I am using c++ 7.5.0)

**Comment by jcarpent** [comment]
@cmastalli Could you provide the `gdb` output with the fix I've proposed of the Python script. It should have changed with the fixed I've proposed.

**Comment by cmastalli** [comment]
> @cmastalli Could you provide the `gdb` output with the fix I've proposed of the Python script. It should have changed with the fixed I've proposed.

Indeed, we have another problem. This is the new output from `gdb`:
```bash
(gdb) run test.py 
Starting program: /usr/bin/python test.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff3dd9700 (LWP 26742)]
[New Thread 0x7ffff35d8700 (LWP 26743)]
[New Thread 0x7fffeedd7700 (LWP 26744)]
[New Thread 0x7fffee5d6700 (LWP 26745)]
[New Thread 0x7fffe9dd5700 (LWP 26746)]
[New Thread 0x7fffe75d4700 (LWP 26747)]
[New Thread 0x7fffe6dd3700 (LWP 26748)]
[New Thread 0x7fffe25d2700 (LWP 26749)]
[New Thread 0x7fffdfdd1700 (LWP 26750)]
[New Thread 0x7fffdd5d0700 (LWP 26751)]
[New Thread 0x7fffdadcf700 (LWP 26752)]
[New Thread 0x7fffd85ce700 (LWP 26753)]
[New Thread 0x7fffd5dcd700 (LWP 26754)]
[New Thread 0x7fffd35cc700 (LWP 26755)]
[New Thread 0x7fffd2dcb700 (LWP 26756)]

Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007fffcb8ca8ca in pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl>::DataTpl(pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> const&) ()
   from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
(gdb) bt
#0  0x00007fffcb8ca8ca in pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl>::DataTpl(pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> const&) ()
   from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
#1  0x00007fffcb8ccadc in boost::python::objects::make_holder<1>::apply<boost::python::objects::value_holder<pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl> >, boost::mpl::vector1<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::execute(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>) ()
   from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
#2  0x00007fffcb89d938 in boost::python::detail::caller_arity<2u>::impl<void (*)(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>), boost::python::default_call_policies, boost::mpl::vector3<void, _object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::operator()(_object*, _object*) [clone .isra.16516] ()
   from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
#3  0x00007fffca323e55 in boost::python::objects::function::call(_object*, _object*) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#4  0x00007fffca324008 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#5  0x00007fffca32abcb in boost::python::detail::exception_handler::operator()(boost::function0<void> const&) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#6  0x00007fffca7bc6d3 in boost::detail::function::function_obj_invoker2<boost::_bi::bind_t<bool, boost::python::detail::translate_exception<eigenpy::Exception, void (*)(eigenpy::Exception const&)>, boost::_bi::list3<boost::arg<1>, boost::arg<2>, boost::_bi::value<void (*)(eigenpy::Exception const&)> > >, bool, boost::python::detail::exception_handler const&, boost::function0<void> const&>::invoke(boost::detail::function::function_buffer&, boost::python::detail::exception_handler const&, boost::function0<void> const&) () from /usr/local/lib/libeigenpy.so
#7  0x00007fffca32a96f in boost::python::handle_exception_impl(boost::function0<void>) () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#8  0x00007fffca3216c9 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#9  0x000055555567a70e in ?? ()
#10 0x000055555567a33a in ?? ()
#11 0x00005555556370ab in ?? ()
#12 0x000055555564db20 in PyEval_EvalFrameEx ()
#13 0x00005555556462aa in PyEval_EvalCodeEx ()
#14 0x0000555555645d19 in PyEval_EvalCode ()
#15 0x000055555567694f in ?? ()
#16 0x0000555555671be2 in PyRun_FileExFlags ()
#17 0x000055555567160d in PyRun_SimpleFileExFlags ()
#18 0x0000555555620356 in Py_Main ()
#19 0x00007ffff7a05b97 in __libc_start_main (main=0x55555561fd30 <main>, argc=2, argv=0x7fffffffdaa8, init=<optimised out>, fini=<optimised out>, rtld_fini=<optimised out>, stack_end=0x7fffffffda98)
    at ../csu/libc-start.c:310
#20 0x000055555561fc4a in _start ()
```

**Comment by cmastalli** [comment]
> > @cmastalli Could you provide the `gdb` output with the fix I've proposed of the Python script. It should have changed with the fixed I've proposed.
> 
> Indeed, we have another problem. This is the new output from `gdb`:
> 
> ```shell
> (gdb) run test.py 
> Starting program: /usr/bin/python test.py
> [Thread debugging using libthread_db enabled]
> Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
> [New Thread 0x7ffff3dd9700 (LWP 26742)]
> [New Thread 0x7ffff35d8700 (LWP 26743)]
> [New Thread 0x7fffeedd7700 (LWP 26744)]
> [New Thread 0x7fffee5d6700 (LWP 26745)]
> [New Thread 0x7fffe9dd5700 (LWP 26746)]
> [New Thread 0x7fffe75d4700 (LWP 26747)]
> [New Thread 0x7fffe6dd3700 (LWP 26748)]
> [New Thread 0x7fffe25d2700 (LWP 26749)]
> [New Thread 0x7fffdfdd1700 (LWP 26750)]
> [New Thread 0x7fffdd5d0700 (LWP 26751)]
> [New Thread 0x7fffdadcf700 (LWP 26752)]
> [New Thread 0x7fffd85ce700 (LWP 26753)]
> [New Thread 0x7fffd5dcd700 (LWP 26754)]
> [New Thread 0x7fffd35cc700 (LWP 26755)]
> [New Thread 0x7fffd2dcb700 (LWP 26756)]
> 
> Thread 1 "python" received signal SIGSEGV, Segmentation fault.
> 0x00007fffcb8ca8ca in pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl>::DataTpl(pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> const&) ()
>    from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
> (gdb) bt
> #0  0x00007fffcb8ca8ca in pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl>::DataTpl(pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> const&) ()
>    from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
> #1  0x00007fffcb8ccadc in boost::python::objects::make_holder<1>::apply<boost::python::objects::value_holder<pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl> >, boost::mpl::vector1<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::execute(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>) ()
>    from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
> #2  0x00007fffcb89d938 in boost::python::detail::caller_arity<2u>::impl<void (*)(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>), boost::python::default_call_policies, boost::mpl::vector3<void, _object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::operator()(_object*, _object*) [clone .isra.16516] ()
>    from /home/cmastalli/devel/pinocchio/build_debug/bindings/python/pinocchio/libpinocchio_pywrap.so
> #3  0x00007fffca323e55 in boost::python::objects::function::call(_object*, _object*) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
> #4  0x00007fffca324008 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
> #5  0x00007fffca32abcb in boost::python::detail::exception_handler::operator()(boost::function0<void> const&) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
> #6  0x00007fffca7bc6d3 in boost::detail::function::function_obj_invoker2<boost::_bi::bind_t<bool, boost::python::detail::translate_exception<eigenpy::Exception, void (*)(eigenpy::Exception const&)>, boost::_bi::list3<boost::arg<1>, boost::arg<2>, boost::_bi::value<void (*)(eigenpy::Exception const&)> > >, bool, boost::python::detail::exception_handler const&, boost::function0<void> const&>::invoke(boost::detail::function::function_buffer&, boost::python::detail::exception_handler const&, boost::function0<void> const&) () from /usr/local/lib/libeigenpy.so
> #7  0x00007fffca32a96f in boost::python::handle_exception_impl(boost::function0<void>) () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
> #8  0x00007fffca3216c9 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
> #9  0x000055555567a70e in ?? ()
> #10 0x000055555567a33a in ?? ()
> #11 0x00005555556370ab in ?? ()
> #12 0x000055555564db20 in PyEval_EvalFrameEx ()
> #13 0x00005555556462aa in PyEval_EvalCodeEx ()
> #14 0x0000555555645d19 in PyEval_EvalCode ()
> #15 0x000055555567694f in ?? ()
> #16 0x0000555555671be2 in PyRun_FileExFlags ()
> #17 0x000055555567160d in PyRun_SimpleFileExFlags ()
> #18 0x0000555555620356 in Py_Main ()
> #19 0x00007ffff7a05b97 in __libc_start_main (main=0x55555561fd30 <main>, argc=2, argv=0x7fffffffdaa8, init=<optimised out>, fini=<optimised out>, rtld_fini=<optimised out>, stack_end=0x7fffffffda98)
>     at ../csu/libc-start.c:310
> #20 0x000055555561fc4a in _start ()
> ```

**Comment by jcarpent** [comment]
I don't think you've used the same compilation flags, you should enable the -g option and remove -DNDEBUG.

**Comment by cmastalli** [comment]
> I don't think you've used the same compilation flags, you should enable the -g option and remove -DNDEBUG.

Sorry for this mistake. I am compiling again. Soon I will share the output.

**Comment by jcarpent** [comment]
@cmastalli I've proposed another commit to fix the issue. Could you test it?

**Comment by cmastalli** [comment]
This is the output produced with your last push:
```bash
(gdb) run test.py 
Starting program: /usr/bin/python test.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff3dd9700 (LWP 30523)]
[New Thread 0x7ffff35d8700 (LWP 30524)]
[New Thread 0x7ffff0dd7700 (LWP 30525)]
[New Thread 0x7fffee5d6700 (LWP 30526)]
[New Thread 0x7fffebdd5700 (LWP 30527)]
[New Thread 0x7fffe75d4700 (LWP 30528)]
[New Thread 0x7fffe6dd3700 (LWP 30529)]
[New Thread 0x7fffe45d2700 (LWP 30530)]
[New Thread 0x7fffdfdd1700 (LWP 30531)]
[New Thread 0x7fffdd5d0700 (LWP 30532)]
[New Thread 0x7fffdadcf700 (LWP 30533)]
[New Thread 0x7fffd85ce700 (LWP 30534)]
[New Thread 0x7fffd5dcd700 (LWP 30535)]
[New Thread 0x7fffd35cc700 (LWP 30536)]
[New Thread 0x7fffd2dcb700 (LWP 30537)]
python: /usr/local/include/eigen3/Eigen/src/Core/DenseStorage.h:128: Eigen::internal::plain_array<T, Size, MatrixOrArrayOptions, 32>::plain_array() [with T = double; int Size = 36; int MatrixOrArrayOptions = 0]: Assertion `(internal::UIntPtr(eigen_unaligned_array_assert_workaround_gcc47(array)) & (31)) == 0 && "this assertion is explained here: " "http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.html" " **** READ THIS WEB PAGE !!! ****"' failed.

Thread 1 "python" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7a24801 in __GI_abort () at abort.c:79
#2  0x00007ffff7a1439a in __assert_fail_base (fmt=0x7ffff7b9b7d8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x7fffcacd5940 "(internal::UIntPtr(eigen_unaligned_array_assert_workaround_gcc47(array)) & (31)) == 0 && \"this assertion is explained here: \" \"http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.htm"..., file=file@entry=0x7fffcacd4a18 "/usr/local/include/eigen3/Eigen/src/Core/DenseStorage.h", line=line@entry=128, 
    function=function@entry=0x7fffcacdd180 <Eigen::internal::plain_array<double, 36, 0, 32>::plain_array()::__PRETTY_FUNCTION__> "Eigen::internal::plain_array<T, Size, MatrixOrArrayOptions, 32>::plain_array() [with T = double; int Size = 36; int MatrixOrArrayOptions = 0]") at assert.c:92
#3  0x00007ffff7a14412 in __GI___assert_fail (
    assertion=0x7fffcacd5940 "(internal::UIntPtr(eigen_unaligned_array_assert_workaround_gcc47(array)) & (31)) == 0 && \"this assertion is explained here: \" \"http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.htm"..., file=0x7fffcacd4a18 "/usr/local/include/eigen3/Eigen/src/Core/DenseStorage.h", line=128, 
    function=0x7fffcacdd180 <Eigen::internal::plain_array<double, 36, 0, 32>::plain_array()::__PRETTY_FUNCTION__> "Eigen::internal::plain_array<T, Size, MatrixOrArrayOptions, 32>::plain_array() [with T = double; int Size = 36; int MatrixOrArrayOptions = 0]") at assert.c:101
#4  0x00007fffc9eaf1d1 in Eigen::internal::plain_array<double, 36, 0, 32>::plain_array (this=0x55555614f6d0) at /usr/local/include/eigen3/Eigen/src/Core/DenseStorage.h:128
#5  0x00007fffc9e9c64c in Eigen::DenseStorage<double, 36, 6, 6, 0>::DenseStorage (this=0x55555614f6d0) at /usr/local/include/eigen3/Eigen/src/Core/DenseStorage.h:187
#6  0x00007fffc9e89b6a in Eigen::PlainObjectBase<Eigen::Matrix<double, 6, 6, 0, 6, 6> >::PlainObjectBase (this=0x55555614f6d0) at /usr/local/include/eigen3/Eigen/src/Core/PlainObjectBase.h:484
#7  0x00007fffc9e743a4 in Eigen::Matrix<double, 6, 6, 0, 6, 6>::Matrix (this=0x55555614f6d0) at /usr/local/include/eigen3/Eigen/src/Core/Matrix.h:259
#8  0x00007fffca5c1c6d in pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl>::DataTpl (this=0x55555614f3f0, model=...)
    at /home/cmastalli/devel/pinocchio/build_debug/include/pinocchio/multibody/data.hxx:108
#9  0x00007fffca5bd621 in boost::python::objects::value_holder<pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl> >::value_holder<boost::python::objects::reference_to_value<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > > (this=0x55555614f3d0, self=0x7fffc23966b0, a0=...) at /usr/include/boost/python/object/value_holder.hpp:133
#10 0x00007fffca5b99a1 in boost::python::objects::make_holder<1>::apply<boost::python::objects::value_holder<pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl> >, boost::mpl::vector1<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::execute (p=0x7fffc23966b0, a0=...) at /usr/include/boost/python/object/make_holder.hpp:94
#11 0x00007fffca7b646a in boost::python::detail::invoke<int, void (*)(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>), boost::python::arg_from_python<_object*>, boost::python::arg_from_python<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > > (
    f=@0x5555560dfb28: 0x7fffca5b9924 <boost::python::objects::make_holder<1>::apply<boost::python::objects::value_holder<pinocchio::DataTpl<double, 0, pinocchio::JointCollectionDefaultTpl> >, boost::mpl::vector1<pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::execute(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>)>, ac0=..., ac1=...)
    at /usr/include/boost/python/detail/invoke.hpp:79
#12 0x00007fffca7ac255 in boost::python::detail::caller_arity<2u>::impl<void (*)(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>), boost::python::default_call_policies, boost::mpl::vector3<void, _object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > >::operator() (this=0x5555560dfb28, args_=0x7ffff7efde10)
    at /usr/include/boost/python/detail/caller.hpp:216
#13 0x00007fffca7a230d in boost::python::objects::caller_py_function_impl<boost::python::detail::caller<void (*)(_object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl>), boost::python::default_call_policies, boost::mpl::vector3<void, _object*, pinocchio::ModelTpl<double, 0, pinocchio::JointCollectionDefaultTpl> > > >::operator() (this=0x5555560dfb20, args=0x7ffff7efde10, kw=0x0)
    at /usr/include/boost/python/object/py_function.hpp:38
#14 0x00007fffc51f5e55 in boost::python::objects::function::call(_object*, _object*) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#15 0x00007fffc51f6008 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#16 0x00007fffc51fcbcb in boost::python::detail::exception_handler::operator()(boost::function0<void> const&) const () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#17 0x00007fffc568e6d3 in boost::detail::function::function_obj_invoker2<boost::_bi::bind_t<bool, boost::python::detail::translate_exception<eigenpy::Exception, void (*)(eigenpy::Exception const&)>, boost::_bi::list3<boost::arg<1>, boost::arg<2>, boost::_bi::value<void (*)(eigenpy::Exception const&)> > >, bool, boost::python::detail::exception_handler const&, boost::function0<void> const&>::invoke(boost::detail::function::function_buffer&, boost::python::detail::exception_handler const&, boost::function0<void> const&) () from /usr/local/lib/libeigenpy.so
#18 0x00007fffc51fc96f in boost::python::handle_exception_impl(boost::function0<void>) () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#19 0x00007fffc51f36c9 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1
#20 0x000055555567a70e in ?? ()
#21 0x000055555567a33a in ?? ()
#22 0x00005555556370ab in ?? ()
#23 0x000055555564db20 in PyEval_EvalFrameEx ()
#24 0x00005555556462aa in PyEval_EvalCodeEx ()
#25 0x0000555555645d19 in PyEval_EvalCode ()
#26 0x000055555567694f in ?? ()
#27 0x0000555555671be2 in PyRun_FileExFlags ()
#28 0x000055555567160d in PyRun_SimpleFileExFlags ()
#29 0x0000555555620356 in Py_Main ()
#30 0x00007ffff7a05b97 in __libc_start_main (main=0x55555561fd30 <main>, argc=2, argv=0x7fffffffdaa8, init=<optimised out>, fini=<optimised out>, rtld_fini=<optimised out>, stack_end=0x7fffffffda98)
    at ../csu/libc-start.c:310
#31 0x000055555561fc4a in _start ()

```

**Comment by jcarpent** [comment]
The issue is coming from Boost.Python. 
In the native mode, Eigen is trying to align struct containing Static matrix using 32 or 64 bits alignment (against 16 bits normally, for ABI compatibility). Boost.Python is not able to retrieve by default this alignment. EigenPy uses a helper function for that, but trying to align on 16 bits.

I do not want to spend my time on it, but this is my conclusion on the status of this issue.
It would be rather useful to ask advanced people on the topic, but I think this feature is not useful.

**Pull request closed without merging**